shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/presentation/detail/PlantDetailComponent.kt

```diff
@@ -16,18 +21,21 @@ class PlantDetailComponent(
     private val plantsRepository: PlantsRepository,
     private val onNavigateToBack: () -> Unit
 ) : ComponentContext by componentContext {
-    private val _state = MutableStateFlow(PlantDetailState(plant))
-    val state: StateFlow<PlantDetailState> = _state.asStateFlow()
+    private var favoritePlants: Flow<List<Plant>> =
+        flow { emitAll(plantsRepository.getFavorites()) }
 
-    init {
-        _state.update { it.copy(isFavourite = it.plant.isFavorite) }
-    }
+    private val _state = MutableStateFlow(PlantDetailState(plant))
+    val state: StateFlow<PlantDetailState> =
+        _state.combine(favoritePlants) { state, favourites ->
```

Comment:
instead of this _state.combine(favoritePlants) use it like this -> _state.combine(plantsRepository.getFavorites())

You dont need that extra :) sorry if the I was unclear

Reply:
I also prefer this way but unfortunately it is causing an error because getFavorites() is a suspending function and cannot be directly called from a non-suspending context. So this was a workaround that I found and I separated into a variable for the sake of readability.